### ISSUE TITLE
loop index variable referenced by lambda used for other index variable results in ambiguous targets error

### ISSUE THREAD

**Issue by fridis** [tokiwa-software/fuzion#3128]
this code 
```
test_loop is
  f(a ()->i32) => a.call
  for c in 0..9
      x := f ()->c
  do
    say x
```
results in 
```
 > ./build/bin/fz test_loop.fz

/home/fridi/fuzion/work/test_loop.fz:4:18: error 1: Ambiguous targets found for call to 'c'
      x := f ()->c
-----------------^
Found several possible call targets within the current feature at different levels of outer features:
in 'test_loop.#loop0' found 'test_loop.#loop0.c' defined at /home/fridi/fuzion/work/test_loop.fz:3:7:
  for c in 0..9
------^
and in 'test_loop.#loop0' found 'test_loop.#loop0.c' defined at /home/fridi/fuzion/work/test_loop.fz:3:7:
  for c in 0..9
------^
and in 'test_loop' found 'test_loop.c' defined at /home/fridi/fuzion/work/test_loop.fz:3:7:
  for c in 0..9
------^
To solve this, you may qualify the feature using 'test_loop.#loop0.this.c' or 'test_loop.#loop0.this.c' or 'test_loop.this.c'.

one error.
```
while after moving the code into the loop body
```
test_loop is
  f(a ()->i32) => a.call
  for c in 0..9
  do
    x := f ()->c
    say x
```
this works fine:
```
 > ./build/bin/fz test_loop2.fz
0
1
2
3
4
5
6
7
8
9
```

**Comment by miclill** [comment]
fixed by #3112

**Comment by michaellilltokiwa** [comment]
fixed by #3112
